pvar.c

```diff
@@ -4624,14 +4624,14 @@ void pv_value_destroy(pv_value_t *val)
 int pv_printf_s(struct sip_msg* msg, pv_elem_p list, str *s)
 {
 	static int buf_itr = 0;
-	static char buf[PV_PRINT_BUF_NO][PV_PRINT_BUF_SIZE];
```

Comment:
`buf` *MUST* be static, otherwise you cannot use it after returning from the `pv_printf_s()` function. With your change, the function is totally broken, as the buffer will be lost on the stack after return.